modules/cudf/src/series.ts

```diff
@@ -648,6 +648,65 @@ export class AbstractSeries<T extends DataType = any> {
     return this.__construct(this._col.gather(selection._col));
   }
 
+  /**
+   * perform nLargest or nSmallest operation
+   * @ignore
+   */
+  _nLargestOrSmallest(ascending: boolean, n: number, keep: DuplicateKeepOption) {
+    if (keep == DuplicateKeepOption.first) {
+      return this.sortValues(!ascending).head(n < 0 ? 0 : n);
+    } else if (keep == DuplicateKeepOption.last) {
+      return n <= 0 ? Series.new({type: this.type, data: new Array(0)})
+                    : this.sortValues(ascending).tail(n).reverse();
+    } else {
+      throw new TypeError('keep must be either "first" or "last"');
+    }
+  }
+
+  /**
+   * Returns the n largest element(s).
+   *
+   * @param n The number of values to retrieve.
+   * @param keep Determines whether to keep the first or last of any duplicate values.
+   *
+   * @example
+   * ```typescript
+   * import {Series} from '@rapidsai/cudf';
+   *
+   * const a = Series.new([4, 6, 8, 10, 12, 1, 2]);
+   * const b = Series.new(["foo", "bar", "test"]);
+   *
+   * a.nLargest(); // [12, 10, 8, 6, 4]
+   * b.nLargest(1); // ["test"]
+   * a.nLargest(-1); // []
+   * ```
+   */
+  nLargest(n = 5, keep: DuplicateKeepOption = DuplicateKeepOption.first) {
+    return this._nLargestOrSmallest(true, n, keep);
+  }
```

Comment:
```suggestion
  nLargest(n = 5, keep: keyof typeof DuplicateKeepOption = 'first') {
    return this._nLargestOrSmallest(true, n, DuplicateKeepOption[keep]);
  }
```

Reply:
What's the reasoning behind preferring this over the enum? Keeping it similar to how it's used in python?

Reply:
Yeah that, and to match how we're doing it in other places